snippets/standard/buffers/memory-t/task-returning/task-returning.cs

```diff
@@ -0,0 +1,55 @@
+﻿// <Snippet1>
+using System;
+using System.Buffers;
+using System.IO;
+using System.Threading.Tasks;
+
+public class Example
+{
+    // <Snippet1>
+    // An acceptable implementation.
+    static void Log(ReadOnlyMemory<char> message)
+    {
+        // Run in the background so that we don't block the main thread while performing IO.
+        Task.Run(() => {
+             StreamWriter sw = File.AppendText(@".\input-numbers.dat");
+            sw.WriteLine(message);
+            sw.Flush();
+        });
+    }
+    // </Snippet1>
+
+    // user code
+    public static void Main()
+    {
+        using (var owner = MemoryPool<char>.Shared.Rent())
+        {
+            var memory = owner.Memory;
+            var span = memory.Span;
+            while (true)
+            {
+                int value = Int32.Parse(Console.ReadLine());
+                if (value < 0) 
+                    return;
+
+                int numCharsWritten = ToBuffer(value, span);
+                Log(memory.Slice(0, numCharsWritten));
```

Comment:
If `Log` returns a `Task`, this can be awaited (and main can be an async main method).

Reply:
That won't work, since spans aren't allowed in async methods.